For an unknown reason, python is stuck indefinitely on Popen.wait() after
the proceses has finished. Add an (arbitrary) timeout of 10 seconds.

diff -urN waf-2.0.25/waflib/Utils.py.orig waf-2.0.25/waflib/Utils.py
--- waf-2.0.25/waflib/Utils.py.orig	2023-01-01 14:48:32.898348000 +0100
+++ waf-2.0.25/waflib/Utils.py	2023-02-13 18:46:20.965541100 +0100
@@ -1042,7 +1042,10 @@
 		except OSError:
 			pass
 		else:
-			k.wait()
+			try:
+				k.wait(10)
+			except Exception:
+				pass
 # see #1889
 if (sys.hexversion<0x207000f and not is_win32) or sys.hexversion>=0x306000f:
 	atexit.register(atexit_pool)
